### DISCUSSION TITLE
Example of creating a new document (POST)

### DISCUSSION THREAD

**Discussion by cleoni** [fattureincloud/api#29] [Q&A] [answered]
Salve, vi chiedo per cortesia un chiarimento.
Guardando la documentazione trovo solo esempi di letture di dati (GET) molto semplici i cui dati in input stanno sull'url e non trovo esempi di GET o POST che prevedano la specifica di dati strutturati in input al sistema.
La documentazione dell'SDK produce del codice JSON per esprimere i dati da dare in input al sistema, per esempio:
`{
"data": {
"attachment_token": "YmMyNWYxYzIwMTU3N212ABCDZjZiMzg5OWY0ODNkZDQveXl5LmRvYw"
}}`

ma non trovo alcun esempio di codice PHP che poi utilizzi e spedisca al cloud tali dati di input. 
Nel codice di esempio vedo passare "headers", ma i dati di input non si sa dove e come inviarli.
E' una cosa che forse può sembrare banale ma se non c'è un esempio tocca andare per tentativi e si perde molto tempo.

**Comment by valmoz** [comment, marked as answer]
Buongiorno @cleoni 
può trovare un esempio completo di creazione di una fattura (quindi di un metodo POST in cui si inviano dati nel body della richiesta) [in questa guida](https://developers.fattureincloud.it/docs/guides/invoice-creation/); selezionando PHP troverà esattamente il codice di cui sta parlando.

Inoltre, la [documentazione della SDK](https://developers.fattureincloud.it/docs/sdks/php-sdk/) contiene un ulteriore esempio di POST.

A disposizione per ulteriori domande,
Mauro

**Reply by cleoni** [reply]
Adesso è un più chiaro grazie. 

Purtroppo il tutto si blocca a causa di qualche libreria che non si autocarica:

`Uncaught Error: Class 'CreateIssuedDocumentRequest' not found`

C'è qualche suggerimento che mi potete dare per caricare tutta la libreria che serve? 
Vedo che composer ha già scaricato **vendor/fattureincloud/fattureincloud-php-sdk** ma evidentemente qualcosa manca ancora.

**Reply by cleoni** [reply]
trovato. c'è un refuso in questa pagina:

https://developers.fattureincloud.it/docs/guides/invoice-creation/#-create-the-document

questa che è indicata a me dà errore, nonostante i vare "use" in cima al programma:

// Here we put our invoice in the request object
$create_issued_document_request = new CreateIssuedDocumentRequest;

in una altra pagina invece ho trovato questa con il percorso assoluto e funziona:

// Here we put our invoice in the request object
$create_issued_document_request = new \FattureInCloud\Model\CreateIssuedDocumentRequest;

pare ora tutto a posto. grazie

**Reply by valmoz** [reply]
Ottimo, grazie mille.
Ho cancellato la mia vecchia risposta perché obsoleta.

Mauro
